sw/otbn/crypto/p384_verify.s

```diff
@@ -318,19 +154,19 @@ p384_verify:
   andi      x2, x2, 1
   beq       x2, x0, fail
 
+  /* Compute Solinas constant k for modulus n (we know it is only 191 bits, so
+     no need to compute the high part):
+     w14 <= 2^256 - n[255:0] = (2^384 - n) mod (2^256) = 2^384 - n */
+  bn.sub    w14, w31, w12
+
   /* Compute modular inverse of S
      Note: This can be replaced by the 'mod_inv_n_p384' subroutine at the
            cost of ~60k cycles if reduced code size is targeted */
   /* [w9,w8] <= [w17,w16] <= s^-1  mod n = [w30,w29]^-1 mod [w13,w12] */
-  jal       x1, mod_inv_var
+  jal       x1, mod_inv_n_p384
```

Comment:
Switching from variable-time to constant-time inversion probably has some performance impact; if you have the time, would you mind reproducing the P-384 verify benchmark here to see if the change is significant? https://opentitan.org/book/hw/ip/otbn/doc/otbn_intro.html#benchmark-reproduction

The modular inversion shouldn't be the hot path so hopefully it's not a huge difference Even if it is, let's not block the PR on this, it would just be good to know if we should allocate some time and energy towards a code-size-optimized variable time inversion.

Reply:
> IBut even if variable-time is more performant, wouldn't it be better from a security point of view to always have constant-time operations anyway?

n the verify case, there isn't any SCA concern; since all the inputs are public, any attacker can easily take the same public key and signature and compute whatever intermediate values they want.

I think the performance hit is small enough that we can absorb it in the name of code size for now. We may at some point want to try using a [known carry chain](https://briansmith.org/ecc-inversion-addition-chains-01#p384_scalar_inversion), for example, to get something that's both constant-time and fast, but I think it's probably not the highest priority at the moment and it would take some experimenting to get the code size down for that method (the naive implementation would be pretty long).

Reply:
> n the verify case, there isn't any SCA concern; since all the inputs are public, any attacker can easily take the same public key and signature and compute whatever intermediate values they want.
ahh right, missed that point.

Carry chain sounds good, but before thinking about implementing something like that it would probably make sense to first try to reduce code size in general. For example by changing the pointer referencing style (as we already discussed here: https://github.com/lowRISC/opentitan/pull/20161). Then we would have a little bit more imem margin for the implementation.

Reply:
Wait a second, I just saw that both implementations, `mod_inv_n_p384`as well as `mod_inv_var`, are variable-time.
`mod_inv_n_p384` is somehow just implemented as a standard square-and-multiply algorithm.
Should we better change this to a square-and-always-multiply algorithm?

Reply:
It's variable time with regard to the modulus, which in this case is a constant. In this context, it only needs to be constant-time relative to the base, which it is.

Reply:
Ah okay I see. Thanks for the explanation!